tests/cobs/src/main.c

```diff
@@ -28,33 +32,78 @@ static void roundtrip_test_runner(const void *input, const size_t length)
 	zassert_mem_equal(decoded_buffer, input, length);
 	zassert_equal(decoded_buffer[length], 0xAB);
 
+	uint8_t *const encoded_buffer2 = malloc(encoded_buffer_length);
+	uint8_t *const decoded_buffer2 = malloc(length + 1);
+	memset(encoded_buffer2, 0xAB, encoded_buffer_length);
+	decoded_buffer2[length] = 0xAB;
+
+	const size_t encoded_length2 =
+		cobs_encode_stream_simple(input, length, encoded_buffer2, encoded_buffer_length);
+	if (length == 254 && !memchr(input, 0x00, length)) {
+		/* The normal encoder writes an additional 0x01, if the
+		 * previous code was 0xff and there is no data left.
```

Comment:
this looks like a small bug in the normal encoder

Reply:
it is and I already fixed it. that fix will show up in a future PR.